Mixpanel/MPABTestDesignerConnection.m

```diff
@@ -253,23 +257,71 @@ - (void)webSocket:(MPWebSocket *)webSocket didCloseWithCode:(NSInteger)code reas
-        _recordingView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, mainWindow.frame.size.width, 1.0)];
-        _recordingView.backgroundColor = [UIColor colorWithRed:4/255.0f green:180/255.0f blue:4/255.0f alpha:1.0];
-        [mainWindow addSubview:_recordingView];
-        [mainWindow bringSubviewToFront:_recordingView];
+        self.connectivityIndicatorWindow = [[UIWindow alloc] initWithFrame:CGRectMake(0, 0, mainWindow.frame.size.width, 4.f)];
+        self.connectivityIndicatorWindow.backgroundColor = [UIColor clearColor];
+        self.connectivityIndicatorWindow.windowLevel = UIWindowLevelAlert;
+        self.connectivityIndicatorWindow.alpha = 0;
+        self.connectivityIndicatorWindow.hidden = NO;
+        
+        _recordingView = [[UIView alloc] initWithFrame:self.connectivityIndicatorWindow.frame];
+        _recordingView.backgroundColor = [UIColor clearColor];
+        _indeterminateLayer = [CALayer layer];
+        _indeterminateLayer.backgroundColor = [UIColor colorWithRed:1/255.0 green:179/255.0 blue:109/255.0 alpha:1.0].CGColor;
+        _indeterminateLayer.frame = CGRectMake(0, 0, 0, 4.0f);
+        [_recordingView.layer addSublayer:_indeterminateLayer];
+        [self.connectivityIndicatorWindow addSubview:_recordingView];
+        [self.connectivityIndicatorWindow bringSubviewToFront:_recordingView];
+        
+        [UIView animateWithDuration:0.3 animations:^{
+            self.connectivityIndicatorWindow.alpha = 1;
+        }];
     }
+    [self animateConnecting:isLoading];
 }
 
-- (void)hideConnectedView
-{
-    if (_recordingView) {
+- (void)animateConnecting:(BOOL)isLoading {
+    if (isLoading) {
+        counter = 0.f;
+        counterIncrease = 0.01f;
+        timer = [NSTimer scheduledTimerWithTimeInterval: 0.1f
+                                                 target: self
+                                               selector: @selector(updateTimer)
+                                               userInfo: nil
+                                                repeats: YES];
+    } else {
+        counterIncrease = 0.1f;
+    }
+}
+
+- (void)updateTimer {
+
+    CGFloat newWidth = MIN(counter + (1- sin(counter*M_PI*2+M_PI/2))/8, 0.95f);
+    if (counterIncrease > 0.01f && newWidth >= 0.95f) {
+        newWidth = 1.f;
+    }
+    
+    [UIView animateWithDuration:0.1f animations:^{
+        self->_indeterminateLayer.frame = CGRectMake(0, 0, self->_recordingView.frame.size.width * newWidth, 4.0f);
+    }];
+    
+    counter += counterIncrease;
+    
+    if (counter > 1 && newWidth >= 1.f) {
+        [timer invalidate];
+    }
+}
+
+- (void)hideConnectedView {
+    if (self.connectivityIndicatorWindow) {
+        [_indeterminateLayer removeFromSuperlayer];
         [_recordingView removeFromSuperview];
+        self.connectivityIndicatorWindow.hidden = YES;
     }
     _recordingView = nil;
+    self.connectivityIndicatorWindow = nil;
```

Comment:
Why do we `nil` this out? Doesn't this mean we have to re-create the view every time it is hidden or shown?